charts/datadog/templates/agent-cilium-network-policy.yaml

```diff
@@ -108,6 +115,52 @@ specs:
           - ports:
               - port: "10250"
                 protocol: TCP
+              - port: "6942"
+                protocol: "TCP"
+  - description: "Egress to Prometheus metrics"
+    endpointSelector:
+      matchLabels:
+        app: {{ template "datadog.fullname" . }}
+        {{- if .Values.agents.podLabels }}
+        {{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+    egress:
+      - toEntities:
+          - host
+        toPorts:
+          - ports:
+              - port: "9090"
+                protocol: "TCP"
+  - description: "Egress to cluster agent"
+    endpointSelector:
+      matchLabels:
+        app: {{ template "datadog.fullname" . }}
+        {{- if .Values.agents.podLabels }}
+        {{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+    egress:
+      - toEndpoints:
+          - matchLabels:
+              app: {{ template "datadog.fullname" . }}-cluster-agent
+        toPorts:
+          - ports:
+              - port: "5005"
+                protocol: TCP
+  - description: "Egress to kube-state-metrics"
+    endpointSelector:
+      matchLabels:
+        app: {{ template "datadog.fullname" . }}
+        {{- if .Values.agents.podLabels }}
+        {{ toYaml .Values.agents.podLabels | indent 8 }}
+        {{- end }}
+    egress:
+      - toEndpoints:
+          - matchLabels:
+              name: kube-state-metrics
+        toPorts:
+          - ports:
+              - port: "8080"
+                protocol: TCP
```

Comment:
`kube-state-metrics` is only one of the numerous checks that a node agent can run.
There is currently no mechanism to dynamically open or close ports in the NetworkPolicy based on the checks that are scheduled by the auto-discovery.
That’s why there’s already [an “allow all egress traffic” rule here](https://github.com/DataDog/helm-charts/blob/7fb6a6ea7e079778d403ae284f3f1f45eaf3b1ac/charts/datadog/templates/agent-cilium-network-policy.yaml#L143-L162).

Rather than introducing a `kube-state-metrics` specific rule, I’d be in favour of fixing the “allow all egress” rule if it’s currently broken.